pkg/proxy/backend/authenticator.go

```diff
@@ -90,8 +92,16 @@ func (auth *Authenticator) handshakeFirstTime(clientIO, backendIO *pnet.PacketIO
 		return err
 	}
 	// Always upgrade TLS with the server.
-	auth.backendTLSConfig = backendTLSConfig
-	if err = backendIO.UpgradeToClientTLS(backendTLSConfig); err != nil {
+	auth.backendTLSConfig = backendTLSConfig.Clone()
+	addr := backendIO.RemoteAddr().String()
+	if auth.serverAddr != "" {
+		addr = auth.serverAddr
+	}
```

Comment:
What's the difference of `backendIO.RemoteAddr().String()` and `auth.serverAddr`?

Reply:
The former will likely be an IP address, will cause problems if your certs are signed by domain.

Reply:
Please add a comment for this.

Reply:
Done.